cms/djangoapps/contentstore/helpers.py

```diff
@@ -616,6 +630,24 @@ def _import_file_into_course(
         if thumbnail_content is not None:
             content.thumbnail_location = thumbnail_location
         contentstore().save(content)
+        if usage_key.block_type == 'video':
+            # Adding transcripts to VAL using the new edx_video_id
+            language_code = next((k for k, v in block.transcripts.items() if v == filename), None)
+            if language_code:
+                sjson_subs = Transcript.convert(
+                    content=data,
+                    input_format=Transcript.SRT,
+                    output_format=Transcript.SJSON
+                ).encode()
+                create_or_update_video_transcript(
+                    video_id=block.edx_video_id,
+                    language_code=language_code,
+                    metadata={
+                        'file_format': Transcript.SJSON,
+                        'language_code': language_code
+                    },
+                    file_data=ContentFile(sjson_subs),
+                )
```

Comment:
This new code does not seem to match the docstring of the function: "Import a single staged static asset file into the course, unless it already exists." It also doesn't use `staged_content_id` nor `file_data_obj`.

I think the code is fine but it should be moved out of `_import_file_into_course` and into a new helper function like `_import_transcripts`

Reply:
Updated https://github.com/openedx/edx-platform/pull/36058/commits/34e5cbc493bf116a18d3e22ed139b3866e230161